datatypes/src/primitives/datetime.rs

```diff
@@ -0,0 +1,668 @@
+                })
+            }
+            (false, true) => Err(DateTimeError::CannotParseOnlyDateWithTimeZone),
+            (false, false) => {
+                let datetime = chrono::NaiveDate::parse_from_str(input, format._to_parse_format())
+                    .map_err(|e| DateTimeError::DateParse {
+                        source: Box::new(e),
+                    })?
+                    .and_hms(0, 0, 0);
+
+                Ok(Self {
+                    datetime,
+                    offset: UTC,
+                })
+            }
+        }
+    }
+
+    pub fn parse_from_rfc3339(input: &str) -> Result<Self, DateTimeError> {
+        let date_time = chrono::DateTime::<chrono::FixedOffset>::parse_from_rfc3339(input)
+            .map_err(|e| DateTimeError::DateParse {
+                source: Box::new(e),
+            })?;
+
+        Ok(date_time.into())
+    }
+
+    // TODO: is this inprecise?
+    pub fn parse_relaxed(input: &str) -> Result<Self, DateTimeError> {
+        let date_time = chrono::DateTime::<chrono::FixedOffset>::from_str(input).map_err(|e| {
+            DateTimeError::DateParse {
+                source: Box::new(e),
+            }
+        })?;
+
+        Ok(date_time.into())
+    }
+
+    pub fn format(&self, format: &DateTimeParseFormat) -> String {
+        let chrono_date_time: chrono::DateTime<chrono::FixedOffset> = self.into();
+        let parse_format = format._to_parse_format();
+
+        chrono_date_time.format(parse_format).to_string()
+    }
+
+    pub fn to_rfc3339(self) -> String {
+        let chrono_date_time: chrono::DateTime<chrono::FixedOffset> = self.into();
+        chrono_date_time.to_rfc3339()
+    }
+
+    /// Now in UTC.
+    pub fn now() -> Self {
+        chrono::offset::Utc::now().into()
+    }
+
+    pub fn day_of_year(&self) -> u16 {
+        // TODO: calculate offset?
+        self.datetime.ordinal() as u16
+    }
+
+    pub fn year(&self) -> i32 {
+        self.datetime.year()
+    }
+    pub fn month(&self) -> u8 {
+        self.datetime.month() as u8
+    }
+
+    /// Zero-based month from 0-11.
+    pub fn month0(&self) -> u8 {
+        self.month() - 1
+    }
+
+    pub fn day(&self) -> u8 {
+        self.datetime.day() as u8
+    }
+    pub fn hour(&self) -> u8 {
+        self.datetime.hour() as u8
+    }
+    pub fn minute(&self) -> u8 {
+        self.datetime.minute() as u8
+    }
+    pub fn second(&self) -> u8 {
+        self.datetime.second() as u8
+    }
+    pub fn millisecond(&self) -> u16 {
+        self.datetime.timestamp_subsec_millis() as u16
+    }
+
+    /// Creates a new `DateTime` from the original one by changing the year.
+    ///
+    /// Returns `None` if the year overflows the date.
+    pub fn with_year(&self, year: i32) -> Option<Self> {
+        Some(Self {
+            datetime: self.datetime.with_year(year)?,
+            offset: self.offset,
+        })
+    }
+}
+
+impl UtcOffset {
+    fn _to_fixed_offset(self) -> chrono::FixedOffset {
```

Comment:
why do the names start with an underscore?

Reply:
I essentially want to prevent that no one just writes pub before that and uses that externally.